consensus/src/experimental/buffer_manager.rs

```diff
@@ -150,6 +156,72 @@ impl StateManager {
         Ok(())
     }
 
+    fn try_advance_executed_root(&mut self) {
+        let mut cursor = self.execution_root.clone();
+        while cursor.is_some() {
+            {
+                let buffer_item = get_elem(&cursor);
+                if !matches!(&*buffer_item, BufferItem::Executed(_)) {
+                    break;
+                }
+            }
+            cursor = get_next(&cursor);
+        }
+        self.execution_root = cursor;
```

Comment:
the root can be None? how does it work if it's None and there's new item coming in?

Reply:
When the roots are none, we set it to be the new item if a new item is pushed.